user/marketo/marketo.go

```diff
@@ -128,22 +129,35 @@ func Client(miniconfig minimarketo.ClientConfig) (minimarketo.Client, error) {
 	return client, nil
 }
 
+// repairManager recreat
+func (m *Connector) repairManager() {
+	miniconfig := Miniconfig(m.config)
+	client, _ := Client(miniconfig)
+	m.client = client
+}
+
 // NewManager creates a new manager based off of input arguments
-func NewManager(logger *log.Logger, config *Config, client minimarketo.Client) (Manager, error) {
+func NewManager(logger *log.Logger, config *Config) (Manager, error) {
+	connector := Connector{
+		logger: logger,
+		config: *config,
+	}
+	if err := config.Validate(); err != nil {
+		return &connector, fmt.Errorf("marketo: config is not valid; %s", err)
```

Comment:
I suggest adding counters for the different error conditions encountered in this method.